data/contracts_finder/processing/scripts/cf_notices_ocds_api.py

```diff
@@ -0,0 +1,416 @@
+            logger.info("hitsCount %s", count)
+        max_page = response.get("maxPage")
+
+        results = response.get('results')
+
+        for notice in results:
+            ocid = notice["releases"][0]["ocid"]
+            if ocid == 'ocds-b5fd17-':
+                continue
+
+            yield notice
+        page += 1
+        ocds_api_params["page"] = page
+
+
+class CFDownload(object):
+    def __init__(self, **kwargs):
+
+        self.session = session_with_backoff()
+        self.too_few_suppliers_skipped = 0
+        self.too_few_matched_suppliers_skipped = 0
+        self.inserted = 0
+        self.fromDate = None
+        self.toDate = None
+        self.saved_files = kwargs.get('update')
+        self.dataset = kwargs.get('dataset')
+
+        if kwargs.get('days') != None:
+            self.fromDate = str(date.today() - timedelta(days=kwargs['days']))
+            self.toDate = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+        else:
+            if kwargs.get('fromdate') != None:
+                self.fromDate = kwargs['fromdate']
+            else:
+                self.fromDate = str(date.today() - timedelta(days=7))
+            if kwargs.get('todate') != None:
+                self.toDate = kwargs['todate']
+            else:
+                self.toDate = str(date.today())
+
+        if self.dataset == 'bodsmatch':
+            self.dataset_dir = 'json_1_1_bods_match'
+        elif self.dataset == 'suppliermatch':
+            self.dataset_dir = 'json_1_1_supplier_ids_match'
+        elif self.dataset == 'raw':
+            self.dataset_dir = 'json_1_1_raw'
+
+    def get_notices(self, datasetpth='json_1_1_bods_match'):
+        """ Get notices from Contracts Finder API
+            With option of searching date periods
+            or searching existing OCIDs that have a sufficient number of matched suppliers.
+         """
+        session = session_with_backoff()
+        if self.saved_files:
+            dirpath = os.path.join(OCDS_OUTPUT_DIR, datasetpth)
+            for filename in os.listdir(dirpath):
+                filepath = os.path.join(dirpath, filename)
+                with open(filepath) as notice:
+                    resp = json.load(notice)
+                    ocds = json.loads(session.get(resp['uri']).content)
+                    yield ocds
+        else:
+
+            notices = search(publishedFrom=self.fromDate, publishedTo=self.toDate)
+            for n in notices:
+                yield n
+
+
+def fix_cf_supplier_ids(json):
+    """ Modify Contracts Finder OCDS JSON enough to be processed """
+    try:
+        # Fixing to match current version, to be processed by ocdskit
+        json['version'] = '1.0'
+        # Giving suppliers distinct IDs (source sets all supplier IDs to zero)
+        for i, supplier in enumerate(json['releases'][0]['awards'][0]['suppliers']):
+            json['releases'][0]['awards'][0]['suppliers'][i]['id'] = str(i)
+    except:
```

Comment:
It's best to avoid general `except:` if possible and also handle the error in the call to the function, rather than the function itself.